dex/src/main/java/io/crate/data/BiArrayRow.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.data;
+
+/**
+ * A {@link ArrayRow} variant which uses 2 backing arrays which can be changed.
+ */
+public class BiArrayRow implements Row {
```

Comment:
This is similar to `CombinedRow` (which requires two backing Row instances, which could be `RowN` or `ArrayRow` to accomplish the same), maybe that could be utilized instead.

Also fine for me to keep the extra instance if you prefer it that way.

Reply:
Yes right, this impl would safe the additional `Row` instances. No strong opinion here, shall I use `CombinedRow` instead?

Reply:
Also no strong opinion, both is fine I think.